### PR TITLE
Update diagnostics

### PR DESCRIPTION
**Pull request by ahartikainen**
## Description

Let's update our diagnostics to be synced with R `posterior` package.

https://github.com/stan-dev/posterior/blob/master/R/convergence.R

This means, we need to 

- fix off-by-one error in `mcse_quantile`.
- update rankdata transformation

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) PR format
- [x] Includes a sample plot to visually illustrate the changes (only for plot-related functions)
- [x] New features are properly documented (with an example if appropriate)?
- [x] Includes new or updated tests to cover the new feature
- [x] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

**Comment by ahartikainen** [comment]
The difference after the change is minimal

(normal random data N=100 --> 4.9515613831374594e-11)

**Comment by ahartikainen** [comment]
I will update the reference diagnostics to use `posterior` package.

File: arviz/stats/diagnostics.py [hunk 1]
```diff
@@ -530,6 +535,13 @@ def _bfmi(energy):
     return num / den
 
 
+def _backtransform_ranks(arr, c=3 / 8):
+    """Backtransformation of ranks."""
```
Comment (canyon289) [inline arviz/stats/diagnostics.py:539]: What does backtransformation do?

Comment (ahartikainen) [reply arviz/stats/diagnostics.py:539]: magic

Comment (ahartikainen) [reply arviz/stats/diagnostics.py:539]: I will update the docstring based on the posterior docstring.

File: arviz/stats/diagnostics.py [hunk 2]
```diff
@@ -530,6 +535,25 @@ def _bfmi(energy):
...
 
+def _backtransform_ranks(arr, c=3 / 8):
+    """Backtransformation of ranks.
+    
+    Parameters
+    ----------
+    arr : np.ndarray
+        Ranks array
+    c : float
+        Fractional offset. Defaults to c = 3/8 as recommended by Bloom (1985).
```
Comment (ahartikainen) [inline arviz/stats/diagnostics.py:546]: @avehtari any idea what paper is this referring to?

Comment (avehtari) [reply arviz/stats/diagnostics.py:546]: There are two typos in the reference. It should be Blom (1958). The reference can be found in the Rhat paper

Blom, G. (1958). Statistical Estimates and Transformed Beta-Variables. Wiley; New York.

Comment (ahartikainen) [reply arviz/stats/diagnostics.py:546]: Thanks

**Review by OriolAbril — APPROVED** [review]

File: arviz/stats/diagnostics.py [hunk 3]
```diff
@@ -536,7 +536,19 @@ def _bfmi(energy):
 
 
 def _backtransform_ranks(arr, c=3 / 8):
-    """Backtransformation of ranks."""
+    """Backtransformation of ranks.
```
Comment (canyon289) [inline arviz/stats/diagnostics.py:539]: I guess dumb question still but what does this actually do again? Either math or practical wise?

File: arviz/tests/base_tests/test_diagnostics.py [hunk 4]
```diff
@@ -54,64 +54,79 @@ def test_bfmi_dataset_bad(self):
...
+                          "mcse_median",
                           "mcse_quantile01",
                           "mcse_quantile10",
                           "mcse_quantile30")
-        write.csv(df, "reference_values.csv")
+        write.csv(df, "reference_posterior.csv")
         ```
+        Reference file:
+
+        Created: 2020-08-31
```
Comment (canyon289) [inline arviz/tests/base_tests/test_diagnostics.py:113]: Very detailed

File: arviz/stats/diagnostics.py (hunk 3 above)
Comment (ahartikainen) [reply arviz/stats/diagnostics.py:539]: old normalization

    (rank - 0.5) / size)

new normalization (c=3/8)

    (rank- c) / (size - 2 * c + 1)

So with c=4/8 they are the same

Comment (avehtari) [reply arviz/stats/diagnostics.py:539]: According to Blom (1958), using fraction 3/8 gives values that match better the expected ordered statistics. This probably doesn't have any practical significance for convergence diagnostics but as the paper reviewer was not familiar with rank normalization, we had to go with the fraction that has been agreed to be the most accurate one.

**Pull request merged**